src/main/java/com/stuypulse/robot/subsystems/ColorSensor.java

```diff
@@ -91,6 +91,21 @@ public CurrentBall getUpdateFromDriverStation() {
                             "DriverStation.getAlliance() returned invalid!", true);
                 }
 
+                return getAllianceColorFromSmartDashboard();
```

Comment:
it looks like you need to implement slightly different logic 


What we need is:
if driverstation alliance color is wrong (it's wrong, not invalid) (this is the case we were warned about), use the network alliance color. We do no not know if the alliance color is wrong in code (hence its spot on shuffleboard) 

What is implemented: 
if driverstation alliance color is invalid (which idk when this happens), use network alliance color 

It woud make sense to cover both, but idk if they would coexist nicely. We definitely want the first one just in case.